pkg/action/install.go

```diff
@@ -750,7 +750,8 @@ func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (
 		return "", err
 	}
 
-	filename, _, err := dl.DownloadTo(name, version, settings.RepositoryCache)
+	filename, rURL, _, err := dl.DownloadTo(name, version, settings.RepositoryCache)
+	c.RepoURL = rURL
```

Comment:
The RepoURL should be reset at upgrade time each time. This is because the chart location can change. For example, let's say I install wordpress from the bitnami repo. Then I decide to vendor it into my own repo. From my own repo I perform an upgrade. The upgraded release should show my vendor repo location instead of the original install location.

Reply:
I think I cover that functionality in the `upgrade.go` file here: https://github.com/helm/helm/pull/10369/files#diff-8f7c1d7e2cfeb70c465f36198e54a053fb517420d8647ffaf72a15e5525eb596R155-R157

That being said, if I'm adding a new function and not updating the `DownloadTo` function I think maybe I can do this in a better way that applies to both install/upgrade commands.

Reply:
Apparently that link doesn't do what I hoped it did. Here's the snippet from `cmd/upgrade.go`. Looking at it again I think you're right. My intention was to update the `repoURL` during an upgrade as well, but I think I may have missed that. I'll fix it up.